Script Canvas: Support for selecting asset IDs from nodes

gadams3 opened the pull request:
## What does this PR do?

This change updates script canvas to support picking, assigning, and generating code for asset IDs. This should resolve issues with the material component, decal component, image-based lighting, gradients, and any other component that requires assets to be assigned from script canvas. While users can now select assets, the script canvas asset picker is now completely unfiltered instead of filtered exclusively to dynamic slices. Additional work will be required to provide script canvas with the necessary asset type info in order to properly filter the asset picker.

https://github.com/o3de/o3de/issues/11307

Signed-off-by: gadams3 <guthadam@amazon.com>

## How was this PR tested?

verify that assets can be assigned to the material component from script canvas hand at the materials are applied when scripts are run in game mode.  

![image](https://user-images.githubusercontent.com/82461473/190526810-f0fb1613-c1b2-409d-9c04-bd0c50daf8e5.png)
![image](https://user-images.githubusercontent.com/82461473/190526875-a9209b34-7609-441a-8594-ed755bb05af6.png)
![image](https://user-images.githubusercontent.com/82461473/190526999-5cb930e6-76d4-45ea-a437-7a0a97ac6171.png)
![image](https://user-images.githubusercontent.com/82461473/190527042-99902d85-580b-4be9-9384-5ac82ae7e61f.png)
![image](https://user-images.githubusercontent.com/82461473/190527146-d5d4f53b-88c4-4db8-9f59-d08f9cf05082.png)

onecent1101 commented:
> ## What does this PR do?
> 
> This change updates script canvas to support picking, assigning, and generating code for asset IDs. This should resolve issues with the material component, decal component, image-based lighting, gradients, and any other component that requires assets to be assigned from script canvas. While users can now select assets, the script canvas asset picker is now completely unfiltered instead of filtered exclusively to dynamic slices. Additional work will be required to provide script canvas with the necessary asset type info in order to properly filter the asset picker.
> 
> #11307
> 
> Signed-off-by: gadams3 [guthadam@amazon.com](mailto:guthadam@amazon.com)
> ## How was this PR tested?
> 
> verify that assets can be assigned to the material component from script canvas hand at the materials are applied when scripts are run in game mode.

onecent1101 reviewed (commented):
Can you provide screenshot to show what is the current behavior look like with this change?

gadams3 commented:
Updated PR with default asset handler for asset ID and cleaned up some of the attribute handling between other asset property handlers. Now assets can also be selected from the inspector. This should also be useful for other classes with reflected asset IDs.
![image](https://user-images.githubusercontent.com/82461473/190533521-82505646-07c6-4e25-9b9a-bef5af341466.png)

cgalvan commented on Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyAssetCtrl.cpp:
```diff
@@ -787,13 +787,11 @@ namespace AzToolsFramework
 
         QMessageBox::warning(GetActiveWindow(), tr("Unable to Edit Asset"),
-            tr("No callback is provided and no associated editor could be found."), QMessageBox::Ok, QMessageBox::Ok);
+            QObject::QObject::tr("No callback is provided and no associated editor could be found."), QMessageBox::Ok, QMessageBox::Ok);
```
Nit: Looks like there are two `QObject::QObject::` here?

gadams3 replied:
Oh no. Thanks. I'll update
